doggos/inference/takagi_sugeno_inference_system.py

```diff
@@ -1,6 +1,49 @@
+from typing import Dict, List, Callable
+
+import numpy as np
+
+from doggos.fuzzy_sets.fuzzy_set import MembershipDegree
 from doggos.inference.inference_system import InferenceSystem
+from doggos.knowledge.linguistic_variable import LinguisticVariable
+from doggos.knowledge.clause import Clause
 
 
 class TakagiSugenoInferenceSystem(InferenceSystem):
-    def output(self) -> float:
-        pass
+    def infer(self,
+              defuzzification_method: Callable,
+              features: Dict[Clause, List[MembershipDegree]],
+              measures: Dict[LinguisticVariable, List[float]]) -> list[float]:
+        """
+        Inferences output based on features of given object and measured values of them, using chosen method
+
+        :param defuzzification_method: method of calculating final output.
```

Comment:
Imo replace 'final output' with' inference system output', because its not formal term.